src/main/java/appeng/client/render/TranslatedRenderItem.java

```diff
@@ -0,0 +1,55 @@
+package appeng.client.render;
+
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.entity.RenderItem;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.item.ItemStack;
+
+import org.lwjgl.opengl.GL11;
+
+/**
+ * Uses translations instead of depth test to perform rendering.
+ */
+public class TranslatedRenderItem extends RenderItem {
+
+    @Override
+    public void renderItemOverlayIntoGUI(FontRenderer font, TextureManager texManager, ItemStack stack, int x, int y,
+            String customText) {
+        if (stack != null) {
+            GL11.glPushMatrix();
+            if (stack.stackSize > 1 || customText != null) {
+                GL11.glTranslatef(0.0f, 0.0f, this.zLevel);
+                String s1 = customText == null ? String.valueOf(stack.stackSize) : customText;
+                GL11.glDisable(GL11.GL_LIGHTING);
+                GL11.glDisable(GL11.GL_BLEND);
+                font.drawStringWithShadow(s1, x + 19 - 2 - font.getStringWidth(s1), y + 6 + 3, 16777215);
```

Comment:
Can you give ability to configure position of the text or direct control on applied offsets?

It can be helpful to me to use in Level Maintainer terminal because I need to show at least two numbers for each stack.

Reply:
I don't think this is the right render item you want. This renders overlay at full scale. you probably want `AppEngRenderItem`

Reply:
For better understanding, since I am not very strong in native GL graphics. How does it become clear that it is used for non-scaled text?

I did something similar for my terminal, but I haven’t yet found where exactly the text size is set.

Reply:
Text size is set by glScale. Mc renders all text the same size otherwise